website_product_sale/models/product.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2018 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import api, fields, models
+from odoo.addons.http_routing.models.ir_http import slug
+
+
+class ProductTemplate(models.Model):
+    _name = 'product.template'
+    _inherit = ['product.template',
+                'website.seo.metadata',
+                'website.published.mixin']
+
+    @api.multi
+    def _compute_website_url(self):
+        # If the product cannot be sold, then use the new form view.
+        # Otherwise, use the defult URL from website_sale, which will
+        # redirect the user to the shop.
+        super(ProductTemplate, self)._compute_website_url()
```

Comment:
I would not call super here: is going to call all the overridden versions of this method included your custom one in website_product, for no benefit.

Plus, `sale_ok` comes from `product` module. So, why can't you do this check here? 
https://github.com/Eficent/website/blob/11.0-website_product/website_product/models/product.py#L21
In there you also say the same about not saleable products... :thinking:

Reply:
I prefer to keep this way, if I remove super I should add /shop/ route manually. Otherwise I can't ensure all url will be computed correctly. But I'm not 100% sure. Tests were failing when super is removed.

Reply:
@hveficent pls read my 2nd comment: I don't think you need this override at all.
Isn't sale_ok already present w/out website_sale? Then why do you need this module at all?
If this is not always true, then I'll improve website_product to delegate the check to a separated method and then in this new glue module you just override this check and you won't have to touch website_url at all :wink:

Reply:
I agree with @simahawk I don't see why this can't be made in the other module :slightly_smiling_face: